client/src/components/edit-instance-view/EditInstanceForm.js

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Grid2 from '@mui/material/Unstable_Grid2';
+import textFormatServices from '../../services/textFormat';
+import DynamicTextFieldArray from './DynamicTextFieldArray';
+import StringTextField from './StringTextFiled';
+import NumberTextField from './NumberTextField';
+
+// TODO: form handling 
+// e.g. check that numbers, dates and emails are submited and nothing of the wrong format
+
+const typeData = {
+  fieldId: 'instanceType',
+  fieldLabel: 'Type'
+};
+
+const startDateData = {
+  fieldId: 'startDate',
+  fieldLabel: 'Starting Date'
+};
+
+const endDateData = {
+  fieldId: 'endDate',
+  fieldLabel: 'Ending Date'
+};
+
+const minCreditsData = {
+  fieldId: 'minCredits',
+  fieldLabel: 'Min Credits'
+};
+
+const maxCreditsData = {
+  fieldId: 'maxCredits',
+  fieldLabel: 'Max Credits'
+};
+
+const gradingScaleData = {
+  fieldId: 'gradingScale',
+  fieldLabel: 'Grading Scale'
+};
+
+const teacherData = {
+  fieldId: 'teacher',
+  fieldLabel: 'Teacher of This Instance'
+};
+
+const EditInstanceForm = ({ instance }) => {
+
+  const [courseType, setType]           = useState(instance.type);
+  const [startDate, setStartDate]       = useState(textFormatServices.formatDateToString(instance.startDate));
+  const [endDate, setEndDate]           = useState(textFormatServices.formatDateToString(instance.endDate));
+  const [teachers, setTeachers]         = useState(instance.responsibleTeachers);
+  const [minCredits, setMinCredits]     = useState(instance.courseData.minCredits);
+  const [maxCredits, setMaxCredits]     = useState(instance.courseData.maxCredits);
+  const [gradingScale, setGradingScale] = useState(instance.gradingScale);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    try {
+      const basicInfoObject = ({
+        courseType,
+        startDate,
+        endDate,
+        teachers,
+        credits,
+        gradingScale,
+      });
+      console.log(basicInfoObject);
+      // TODO: possibly save instance information to DB 
+      // (save info at least somewhere until the creation of the instance is done or disrupted)
+    } catch (exception) {
+      console.log(exception);
+    }
+  };
+
+  return(
+    <form onSubmit={handleSubmit}>
+      <Box sx={{ 
+        display: 'flex', 
+        alignItems: 'flex-start',
+        flexDirection: 'column', 
+        justifyContent: 'space-around',
+        boxShadow: 2, 
+        borderRadius: 2,
+        my: 2,
+        p: 2
+      }}>
+        <StringTextField fieldData={typeData} value={courseType} setFunction={setType}/>
+        <Grid2 container spacing={2}>
+          <Grid2 xs={6}><StringTextField fieldData={startDateData} value={startDate} setFunction={setStartDate}/></Grid2>
```

Comment:
While I don't know so much about React, I wonder if there is a way to have something other than a `StringTextField` here. I know that HTML has `<input type="date">`, for instance, what are your thoughts?

Reply:
Good point, already thought about this earlier! Now the input type for dates is 'date' and for credits it is 'number'.